File: trees/102. Binary Tree Level Order Traversal/levelOrder.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if (!root) {
        return [];
    }

    let result = [];
    let currentLevel = [root];

    while (currentLevel.length > 0) {
        currentLevelValues = [];
        nextLevel = [];

        for (let node of currentLevel) {
            currentLevelValues.push(node.val)

            if (node.left) nextLevel.push(node.left);
            if (node.right) nextLevel.push(node.right);
        }

        result.push(currentLevelValues);
        currentLevel = nextLevel;
    }

    return result;
};

File: trees/102. Binary Tree Level Order Traversal/levelOrder.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        
        result = []
        current_level = [root]
        
        while current_level:
            current_level_values = []
            next_level = []
            
            for node in current_level:
                current_level_values.append(node.val)
                
                if node.left:
                    next_level.append(node.left)
                if node.right:
                    next_level.append(node.right)
                
            result.append(current_level_values)
            current_level = next_level
        
        return result